Source/WebCore/page/Navigation.cpp

```diff
@@ -130,8 +134,36 @@ Navigation::Result Navigation::forward(Options&&, Ref<DeferredPromise>&& committ
     return result;
 }
 
-void Navigation::updateCurrentEntry(UpdateCurrentEntryOptions&&)
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-updatecurrententry
+ExceptionOr<void> Navigation::updateCurrentEntry(JSDOMGlobalObject& globalObject, UpdateCurrentEntryOptions&& options)
 {
+    if (!window()->frame() || !window()->frame()->document())
+        return Exception { ExceptionCode::InvalidStateError };
+
+    auto current = currentEntry();
+    if (!current)
+        return Exception { ExceptionCode::InvalidStateError };
+
+    // Not explicitly in the spec, but expected by web-platform-tests/navigation-api/updateCurrentEntry-method/exception-order-initial-about-blank-unserializablestate.html
+    bool isAboutBlank = window()->frame()->document()->url().isAboutBlank();
+
+    auto serializedState = SerializedScriptValue::create(globalObject, options.state, SerializationForStorage::Yes, isAboutBlank ? SerializationErrorMode::NonThrowing : SerializationErrorMode::Throwing);
```

Comment:
Can't we just always use the variant that returns ExceptionOr, and return that if there is an exception? Chromium code does not seem to have special code for about:blank at this level.

Reply:
> Can't we just always use the variant that returns ExceptionOr, and return that if there is an exception?

This is a void function, so I don't think it looks any nicer anyway.

```cpp
    Vector<RefPtr<MessagePort>> dummyPorts;
    auto serializeResult = SerializedScriptValue::create(globalObject, options.state, { }, dummyPorts, SerializationForStorage::Yes);
    if (serializeResult.hasException()) {
        // Not explicitly in the spec, but expected by web-platform-tests/navigation-api/updateCurrentEntry-method/exception-order-initial-about-blank-unserializablestate.html
        if (window()->frame()->document()->url().isAboutBlank())
            return Exception { ExceptionCode::InvalidStateError };
        return serializeResult.exception().isolatedCopy();
    }
```

Reply:
Do you know where in the specification the special behaviour of about:blank is described? I am not convinced yet it should be part of this change.

Reply:
I did not find it in the spec, only in WPT. https://github.com/web-platform-tests/wpt/blob/master/navigation-api/updateCurrentEntry-method/exception-order-initial-about-blank-unserializablestate.html

And there is no descriptive history there of why.

Reply:
I've removed it from this patch for now.